src/components/wizard/wizard.component.spec.ts

```diff
@@ -249,3 +250,107 @@ describe('Wizard with validator', () => {
             .innerText.trim();
     }
 });
+
+interface WizardStep {
+    header: string;
+    content: string;
+}
+
+@Component({
+    selector: 'wizard-visited-change-test-app',
+    template: `
+        <ux-wizard
+            (stepChanging)="stepChanging($event)"
+            (stepChange)="stepChange($event)"
+            (onNext)="onNext($event)"
+        >
+            <ux-wizard-step *ngFor="let step of steps"
+                [header]="step.header" [valid]="step.valid"
+                (visitedChange)="visitedChanged()"
+            >
+                <p>{{ step.content }}</p>
+                <button class="toggle-validity-button" (click)="step.valid = !step.valid">Toggle validity</button>
+            </ux-wizard-step>
+        </ux-wizard>
+    `
+})
+class WizardVisitedChangeTestComponent {
+    steps: WizardStep[] = [
+        {
+            header: '1. First Step',
+            content: 'Content of step 1.',
+        },
+        {
+            header: '2. Second Step',
+            content: 'Content of step 2.',
+        },
+        {
+            header: '3. Third Step',
+            content: 'Content of step 3.',
+        },
+        {
+            header: '4. Fourth Step',
+            content: 'Content of step 4.',
+        }
+    ];
+    stepChanging(_: StepChangingEvent) {}
+    stepChange(_: number) {}
+    onNext(_: number) {}
+    visitedChanged(_: boolean) { }
+}
+
+describe('Wizard with visitedChange event', () => {
+    let component: WizardVisitedChangeTestComponent;
+    let fixture: ComponentFixture<WizardVisitedChangeTestComponent>;
+    let nativeElement: HTMLElement;
+    let visitedChanged: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [WizardModule, CommonModule],
+            declarations: [WizardVisitedChangeTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(async () => {
+        fixture = TestBed.createComponent(WizardVisitedChangeTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        visitedChanged = spyOn(component, 'visitedChanged');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+        await fixture.whenStable();
+    });
+
+    it('should trigger a visitedChange event when valid modified on the current step, when other steps ahead are visited', async () => {
```

Comment:
Could we also add a test perhaps testing programmatically setting the valid state? I know its fairly common for teams to programmatically modify step properties on the wizard so it would probably be good to test it

Reply:
We can ignore this now that we are no longer relying on `ngOnChanges`